### PR TITLE
Unify execution subtypes

### PR DESCRIPTION
**Pull request by robfletcher**
No more `Pipeline` and `Orchestration` subtypes and generic signatures on `Stage`. Everything is now an `Execution` with an enum specifying the subtype.

### CODE REVIEW

**Comment by lwander** [comment]
Hey rob - we're trying to release 1.5.0 today - could you hold of on merging until later this afternoon?

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/PipelineExpressionEvaluator.java [hunk 1]
```diff
@@ -16,19 +16,19 @@
...
+import org.springframework.expression.spel.standard.SpelExpressionParser;
 import org.springframework.expression.spel.support.StandardEvaluationContext;
-
-import java.util.*;
-
-import static com.netflix.spinnaker.orca.pipeline.expressions.PipelineExpressionEvaluator.ExpressionEvaluationVersion.V2;
 import static com.netflix.spinnaker.orca.pipeline.expressions.PipelineExpressionEvaluator.ExpressionEvaluationVersion.V1;
+import static com.netflix.spinnaker.orca.pipeline.expressions.PipelineExpressionEvaluator.ExpressionEvaluationVersion.V2;
+
+;
```
Comment (robzienert) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/expressions/PipelineExpressionEvaluator.java]: Extra `;` floating.

File: orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Execution.java [hunk 2]
```diff
@@ -376,6 +429,8 @@ public long getPausedMs() {
     }
   }
 
+  public enum ExecutionType {pipeline, orchestration}
```
Comment (cfieber) [inline orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Execution.java]: not a fan of these being lowercase

**Comment by cfieber** [comment]
what is the serialization compatibility story with this change and any existing persisted pipelines / orchestrations?

**Comment by lwander** [comment]
Release is done

**Comment by robfletcher** [comment]
No problem. I need to migrate queue messages anyway.

Regards,
Rob Fletcher

**Comment by robfletcher** [comment]
@cfieber no impact to Redis schema as the execution type isn't stored as a field in the hash. Each sub-type is indexed separately. I've included a migrator for queue data.

**Comment by lwander** [comment]
Any special care that needs to be taken when deploying this with live executions?

**Comment by robfletcher** [comment]
@lwander there's a migrator included for queue messages that will run before the queue initializes. Persistence of the pipelines & orchestrations themselves is not changed.

**Comment by lwander** [comment]
Sweet, thanks

**Comment by andrewbackes** [comment]
@robfletcher How long are you planning to leave the migrator in the codebase?

**Pull request merged**